blog/serialize-a-style-using-your-chakra-ui-theme.md

```diff
@@ -0,0 +1,113 @@
+
+And I found myself in an issue with the input for the card number and the CVV number because Spreedly inserts an `iframe` and the inputs are not under my control (and my CSS).
+
+## The easy solution
+
+Afortunately Spreedly accepts a style inline string via the [`setStyle`](https://docs.spreedly.com/reference/iframe/v1/#setstyle) function.
+
+```javascript
+Spreedly.on("ready", () => {
+  Spreedly.setStyle("number", "width:225px;  height:35px;");
+});
+```
+
+A little ugly but a solution at least. Lets see how to improve this...
+
+## Using the theme with strings
+
+We can use the [`useTheme` hook](https://chakra-ui.com/docs/hooks/use-theme) to get all the tokens we had defined in the theme.
+
+```javascript
+import {useEffect} from 'react'
+import {useTheme} from "@chakra-ui/react"
+
+const buildStyles = theme => `
+  border: 2px solid ${theme.colors.gray[300]},
+  color: ${theme.colors.black},
+  lineHeight: ${theme.sizes[5]}
+`
+
+const MyComp = () => {
+  const theme = useTheme();
+  useEffect(() => {
+    Spreedly.on("ready", () => {
+      Spreedly.setStyle("number", buildStyles(theme));
+    });
+  }, []);
+
+  return <>...</>;
+};
+```
+
+This is similar to `styled-components` using the string templates for create styles.
+
+💡 Tip: remember that your component should be wrapper by a `ChakraProvider` to get the `theme` object.
+
+## Using a CSS object
+
+I looked for a nicer way to handle a CSS object in JavaScript instead of a one big string. Chakra UI uses [emotion](https://emotion.sh/) under the hook to build the CSS classes, so I found this solution:
+
+```javascript
+import {css} from "@chakra-ui/react"
+import {serializeStyles} from '@emotion/serialize'
+
+const toCSSString = (styles, theme) => serializeStyles([css(styles)(theme)]).styles;
+```
+
+the [`serializeStyles` function](https://github.com/emotion-js/emotion/blob/main/packages/serialize/) from `emotion` convert an object into another one built with a `name` attribute for a auto-generated CSS class name; and the `styles` attribute with all the style properties in one string. 😁
+
+the [`css` function](https://github.com/chakra-ui/chakra-ui/blob/main/packages/styled-system/src/css.ts) from Chakra UI normalize the shortcuts that [Chakra provides](https://chakra-ui.com/docs/features/style-props) like:
+
+```
+<Box w="full" h={9} bg="blue.300"/>
+```
+
+The `w`, `h` and `bg` are aliases for `width`, `height` and `background` style properties. The `props` for this `Box` component are passed to the `css` getting this output:
+
+```javascript
+{
+  height: "var(--chakra-sizes-9)",
+  background: "var(--chakra-colors-blue-300)",
+​  width: "var(--chakra-sizes-full)"
+}
+```
+
+We can't use nice values like `9`, `full` or `blue.300` because Spreedly is inside an `iframe` and our CSS [custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) (a.k.a. CSS variables) are not in the scope of the `iframe`'s stylesheet.
```

Comment:
```suggestion
Here we can't use nice values like `9`, `full` or `blue.300` because Spreedly is inside an `iframe` and our CSS [custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) (a.k.a. CSS variables) are not in the scope of the `iframe`'s stylesheet.
```